ctest.h

```diff
@@ -391,7 +419,7 @@ static void color_print(const char* color, const char* text) {
         printf("%s\n", text);
 }
 
-#ifdef __APPLE__
+#if defined (__APPLE__) && !defined (_POSIX_C_SOURCE)
```

Comment:
It appears `RTLD_DEFAULT` is only defined if the feature test macro `_GNU_SOURCE` is defined in the source when compiling with `-std=c99`. If both `_GNU_SOURCE` and `_C_POSIX_SOURCE` are defined, it appears the POSIX overrides the GNU extensions that defines `RTLD_DEFAULT` (and also the function `sys_siglist`).